Composer/packages/client/src/pages/botProject/adapters/ABSChannelSpeechModal.tsx

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import React, { useState } from 'react';
+import formatMessage from 'format-message';
+import { DialogFooter } from 'office-ui-fabric-react/lib/Dialog';
+import { PrimaryButton, DefaultButton } from 'office-ui-fabric-react/lib/Button';
+import { DialogWrapper, DialogTypes } from '@bfc/ui-shared';
+import { TextField } from 'office-ui-fabric-react/lib/TextField';
+import { Link } from 'office-ui-fabric-react/lib/Link';
+
+type Props = {
+  isOpen: boolean;
+  onClose: () => void;
+  onUpdateKey: (key: string, region: string, isDefault: boolean) => Promise<void>;
+};
+
+const ABSChannelSpeechModal = (props: Props) => {
+  const { isOpen, onClose, onUpdateKey } = props;
+
+  const [value, setValue] = useState<string | undefined>();
+  const [region, setRegion] = useState<string | undefined>();
+
+  return (
+    <DialogWrapper
+      data-testid={'absChannelsSpeechModal'}
+      dialogType={DialogTypes.Customer}
+      isOpen={isOpen}
+      title={formatMessage('Connect to Speech Service')}
+      onDismiss={onClose}
+    >
+      <div data-testid="absChannelsSpeechModal">
+        <p>
+          {formatMessage('Provide a key in order to connect your bot to the Azure Speech service. ')}
+          <Link
+            href="https://docs.microsoft.com/en-us/azure/cognitive-services/cognitive-services-apis-create-account?tabs=multiservice%2Cwindows"
```

Comment:
nit: put this link into a variable with a name describing what it's about

Reply:
thanks yes.